### PR TITLE
Improve discovery logging a bit.

### PR DESCRIPTION
**Pull request by bzbarsky-apple**
1) In shared "platform" discovery code, log when we discovered an operational id
   or a commissionable instance and what the TXT records were.
2) In Darwin code, do Discovery, not DeviceLayer logging and improve some of the
   log messages.

The intent is to make it a little clearer on failure where in the pipeline
things stop propagating information.

#### Problem
Can't tell whether we never got "all" addresses or whether consumer just ignored the result.

#### Change overview
Log when we plan to notify the consumer.

#### Testing
Ran chip-tool, saw the updated logs.

### CODE REVIEW

File: src/lib/dnssd/Discovery_ImplPlatform.cpp [hunk 1]
```diff
@@ -134,8 +137,11 @@ static void HandleNodeIdResolve(void * context, DnssdService * result, const Spa
     }
     nodeData.resolutionData.numIPs = addressesFound;
 
+    ChipLogDetail(Discovery, "Node with operational ID %s resolved. TXT:", result->mName);
```
Comment (Damian-Nordic) [inline src/lib/dnssd/Discovery_ImplPlatform.cpp:140]: Isn't it displaying almost the same info as `LogNodeIdResolved` below?

File: src/lib/dnssd/Discovery_ImplPlatform.cpp [hunk 2]
```diff
@@ -70,8 +70,11 @@ static void HandleNodeResolve(void * context, DnssdService * result, const Span<
 
     nodeData.resolutionData.port = result->mPort;
 
+    ChipLogDetail(Discovery, "Node with instance name %s discovered. TXT:", nodeData.commissionData.instanceName);
```
Comment (Damian-Nordic) [inline src/lib/dnssd/Discovery_ImplPlatform.cpp:73]: `DiscoveredNodeData` has `logDetail` method which is supposed to print that. Isn't it a duplicate?

**Review by Damian-Nordic — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

File: src/lib/dnssd/Discovery_ImplPlatform.cpp (hunk 1 above)
Comment (bzbarsky-apple) [reply src/lib/dnssd/Discovery_ImplPlatform.cpp:140]: Hmm.  I guess it is.  The latter does not log T (which I can add) and does not log unknown TXT keys, but that's probably OK.

File: src/lib/dnssd/Discovery_ImplPlatform.cpp (hunk 2 above)
Comment (bzbarsky-apple) [reply src/lib/dnssd/Discovery_ImplPlatform.cpp:73]: Yes, thank you!

**Pull request merged**